README.md

```diff
@@ -143,6 +143,24 @@ http://moodle.local/admin/settings.php?section=tool_heartbeat
 * By default in a new install this is set to 'error'. This is done intertionally so that you know your monitoring is wired up correctly end to end. You should see you monitoring raise an alert which tells you that it is a test and links to the admin setting to turn it into normal monitoring mode.
 * Optionaly lock down the endpoints by IP
 
+## Task fail delay maximum alerting level configuration
+This plugin allows configuring maximum permitted check level for faildelay checks for tasks.
+
+You can provide a default for all tasks, as well as per task configurations that have precedence over the task default.
+
+The configuration is stored in the site config.php under the config setting $CFG->tool_heartbeat_tasks
+
+This should be an array with a form like the following
+```
+$CFG->tool_heartbeat_tasks = [
```

Comment:
Broadly I'm good for this to land, few other minor comments below. The one big thing worth asking is why this is task centric and not check centric? If could have almost exactly the same semantics if converted to checks, except it would be much more powerful and cover everything.

So instead it would look like this instead:

```
$CFG->tool_heartbeat_checks = [
	'core_task_delete_unconfirmed_users_task' => [
		'maxresultlevel' => 'critical',
	],
```

Heartbeat automagically converts all tasks into their own check and merges it with all the existing checks, so for example if that task fails then this details page is created dynamically:

https://master.localhost/report/status/index.php?detail=core_task_delete_unconfirmed_users_task

Reply:
yep, that seems reasonable and I can hook `process_check_and_get_result` to achieve this, but what would I use to identify each individual check?

Were you thinking to match against `$check->get_id()` or `$check->get_name()`?

Reply:
It's actually not clear to me if instances of `failingtaskcheck` have a unique id

Reply:
I've reworked the pull request to work on the check level.